### PR TITLE
Lakhveer/updateResize Adding a sub capability named update in dialogs.

### PR DESCRIPTION
**Pull request by lakhveerk**
1. Created a sub capability named update in dialogs. Moved resize functions in there and added isSupported function. 
2. Made changes in tasks.update, so it would keep working as it was before.
3. Fixed Unit testing and test-app file to support the sub capability.

### CODE REVIEW

File: change/@microsoft-teams-js-f4f055db-c85c-4ea2-b470-278759a1aafb.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "major",
+  "comment": "In `dialog.ts` moved `Resize` functionality in a subcapability named `update`.",
```
Comment (aeunms) [inline change/@microsoft-teams-js-f4f055db-c85c-4ea2-b470-278759a1aafb.json:3]: I think it would be good to specify the capability since the client would be calling it from the capability rather than looking at the file name. So maybe something like:

"`dialog.resize()` function has been moved to a new dialog subcapability and is now `dialog.update.resize()`."

Comment (aeunms) [inline change/@microsoft-teams-js-f4f055db-c85c-4ea2-b470-278759a1aafb.json:3]: I think it would be good to specify the capability since the client would be calling it from the capability rather than looking at the file name. So maybe something like:

"`dialog.resize()` function has been moved to a new dialog subcapability and is now `dialog.update.resize()`."

Comment (aeunms) [inline change/@microsoft-teams-js-f4f055db-c85c-4ea2-b470-278759a1aafb.json:3]: Another change that looks like it should be included is that the input type is now different. The parameter has changed from type DialogInfo to DialogSize.

Comment (aeunms) [inline change/@microsoft-teams-js-f4f055db-c85c-4ea2-b470-278759a1aafb.json:3]: Another change that should be included is that the input type is now different. The parameter has changed from type DialogInfo to DialogSize.

File: packages/teams-js/src/public/tasks.ts [hunk 2]
```diff
@@ -85,4 +92,10 @@ export namespace tasks {
       return TaskModuleDimension.Small;
     }
   }
+
+  export function getDefaultSizeIfNotProvided(taskInfo: TaskInfo): TaskInfo {
```
Comment (TrevorJoelHarris) [inline packages/teams-js/src/public/tasks.ts:96]: Same question I left on the other PR: we had mentioned in the doc we showed to partners that we would default to Small. Why the change?

**Review by TrevorJoelHarris — CHANGES_REQUESTED** [review]
Just waiting on discussion about default dialog size, then I will sign off

File: packages/teams-js/src/public/tasks.ts (hunk 2 above)
Comment (lakhveerk) [reply packages/teams-js/src/public/tasks.ts:96]: It was not intentional. Sorry, I missed it from the doc. Will change to small

File: apps/teams-test-app/src/components/DialogAPIs.tsx [hunk 3]
```diff
@@ -68,21 +69,41 @@ const DialogAPIs = (): ReactElement => {
       },
     });
 
+  const UpdateTaskModule = (): ReactElement =>
+    ApiWithTextInput<TaskInfo>({
+      name: 'updateTaskModule',
+      title: 'Update Task Module',
+      onClick: {
```
Comment (jadahiya-MSFT) [inline apps/teams-test-app/src/components/DialogAPIs.tsx:76]: This seems unnecessary since you accomplish the same goal with the `withCallback` call

**Review by jadahiya-MSFT — CHANGES_REQUESTED** [review]
Everything looks good, I just think we can remove `UpdateTaskModule` UI since it works the same as `ResizeDialog`

File: apps/teams-test-app/src/components/DialogAPIs.tsx (hunk 3 above)
Comment (lakhveerk) [reply apps/teams-test-app/src/components/DialogAPIs.tsx]: Deleted. I think it is a little confusing that we are using 'withPromise' for v2 even though, we actually are using callbacks.

**Review by jadahiya-MSFT — APPROVED** [review]

**Comment by lakhveerk** [comment]
> 

I have changed the size to small :) and it is already merged as part of open API

**Review by TrevorJoelHarris — APPROVED** [review]
